src/main/java/org/humancellatlas/ingest/analysis/web/AnalysisController.java

```diff
@@ -72,7 +93,7 @@
     ResponseEntity<Resource<?>> addFileReference(@PathVariable("analysis_id") Analysis analysis,
                                                  @RequestBody File file,
                                                  final PersistentEntityResourceAssembler assembler) {
-        SubmissionEnvelope submissionEnvelope = analysis.getSubmissionEnvelope();
+        SubmissionEnvelope submissionEnvelope = analysis.getLatestSubmissionEnvelope();
         file.addToSubmissionEnvelope(submissionEnvelope);
         File entity = getFileRepository().save(file);
         Analysis result = getAnalysisService().getAnalysisRepository().save(analysis.addFile(entity));
```

Comment:
I can imagine a situation where this doesn't work very well (e.g. if you try to add a file to an analysis whose latest submission is complete).

I *think* the submission envelope must be open to add a file to an analysis - so therefore maybe we should just change this to `POST /submissionEnvelopes/{sub_id}/analyses/{analysis_id}/files`

Reply:
I think the linking of files to analysis shouldn't be affected. This just gets the last in the list, which should be still currently open when linking files to analysis